src/visualization/interpolate.jl

```diff
@@ -250,6 +250,100 @@ function unstructured_2d_to_1d(original_nodes, unstructured_data, nvisnodes, sli
   return get_data_1d(reshape(new_nodes[:, 1:new_id], 1, n_nodes_in, new_id), new_unstructured_data[:, 1:new_id, :], nvisnodes)
 end
 
+# Convert 2d unstructured data to 1d data at given coordinates.
+function unstructured_2d_to_1d_along_curve(original_nodes, unstructured_data, nvisnodes, coordinates, mesh, solver, cache)
+
+  @assert size(coordinates) == (2, size(coordinates)[2]) "Coordinates along curve must be 2xn dimensional."
+
+  n_elements_curve = size(coordinates)[2]
+  n_nodes, _, n_elements, n_variables = size(unstructured_data)
+  nodes_in, _ = gauss_lobatto_nodes_weights(n_nodes)
+
+  # Setup data structures.
+  nodes_on_curve = collect(range(0, 1, length = n_elements_curve))
+  data_on_curve = Array{Float64}(undef, n_elements_curve, n_variables)
+  temp_data = Array{Float64}(undef, n_nodes, n_elements_curve, n_variables)
+
+  # For each coordinate find the corresponding element with its id.
+  element_ids = get_elements_by_coordinates(coordinates, mesh, solver, cache)
+
+  min = original_nodes[:, 1, 1, 1]
+  max = max_coordinate = original_nodes[:, n_nodes, n_nodes, n_elements]
+
+  # Iterate over all found elements.
+  for element in 1:n_elements_curve
+    @assert (prod(vcat(coordinates[:, n_elements_curve] .>= min, coordinates[:, n_elements_curve]
+            .<= max))) "Some coordinates from `along_curve` are outside of the domain.."
+
+    min_coordinate = original_nodes[:, 1, 1, element_ids[element]]
+    max_coordinate = original_nodes[:, n_nodes, n_nodes, element_ids[element]]
+    element_length = max_coordinate - min_coordinate
+
+    normalized_coordinates = (coordinates[:, element] - min_coordinate)/element_length[1]*2 .-1
+
+    # Interpolate to a single point in each element.
+    vandermonde_i = polynomial_interpolation_matrix(nodes_in, normalized_coordinates[1])
+    vandermonde_ii = polynomial_interpolation_matrix(nodes_in, normalized_coordinates[2])
```

Comment:
Instead of `_i` and `_ii`, maybe just call it `_x` and `_y`?

Reply:
Done